.github/workflows/test_on_push.yml

```diff
@@ -32,37 +33,58 @@ jobs:
           python -m pip install pre-commit
           pre-commit run -a
 
-  run_unit_integration_and_coverage_tests:
+  run_windows_unit_integration_tests:
     needs: style
-    runs-on: ${{ matrix.os }}
+    env:
+      BUILD_IDAKLU: OFF
```

Comment:
Have a look this [job](https://github.com/cringeyburger/PyBaMM/actions/runs/10536847176/job/29197492807#step:9:3760). I turned on IDAKLU build for Windows. There was no problem with building it, but the tests failed because of the workers crashing. I tried the same on my Windows laptop, and it's even worse.

Even the current development branch doesn't test for IDAKLU on Windows, so I think we might have to spend a bit more time, and guessing by the amount of work needed, it's better tackled on another fresh PR.

Reply:
Idaklu-jax had an issue that was fixed here #4379 Not sure if that was related.

Reply:
Just enabled IDAKLU in Windows tests

Reply:
Hmm.. There is a problem with the repair step of the Windows CI builds now, caused by the recent updates in the development branch. Here is the [job](https://github.com/cringeyburger/PyBaMM/actions/runs/10567902406/job/29277837156#step:8:748)

Reply:
Am I expected to fix this? Looks hard 😅

Reply:
I was trying to build pybamm locally on windows, and to do that I had to update MSVC 2022. So is it related to that?

Reply:
@agriyakhetarpal Also for the single thread tests, yes it still failed